src/main/java/com/example/backend/timer/OtomotoTimer.java

```diff
@@ -0,0 +1,230 @@
+
+    /**
+     * Run otomoto timer synchronously.
+     * fixedRated = 60 * 5 * 1000 is equal to 5 minutes delay
+     * <p>
+     * Timer fetches new cars for every available filter.
+     * If filter is invalid, then it's being ignored.
+     *
+     * @throws EntityNotFoundException thrown if application setting is missing
+     */
+    @Scheduled(fixedRate = 60 * 5 * 1000)
+    public void runTimer() throws EntityNotFoundException {
+        List<User> allUsers = userService.getAll();
+        if (Collections.isNullOrEmpty(allUsers))
+            return;
+
+        String query = applicationSettingService.getString(Key.OTOMOTO_QUERY_STRING);
+
+        try {
+            query = OBJECT_MAPPER.readValue(query, OtomotoDto.class).getQuery();
+        } catch (JsonProcessingException e) {
+            LOGGER.error("[{}] Invalid query: {}", TIMER_NAME, query);
+            return;
+        }
+
+        Set<Filter> filters = allUsers.stream().flatMap((user) -> user.getFilters().stream()).collect(Collectors.toSet());
+        if (Collections.isNullOrEmpty(filters))
+            return;
+
+        for (Filter filter : filters)
+            loadDataForSingleFilter(filter, query);
+
+        LOGGER.info("[{}][{}] Timer completed.", new SimpleDateFormat("HH:mm:ss").format(new Date()), TIMER_NAME);
+    }
+
+    /**
+     * Prepare dto which will be sent to otomoto REST API
+     * and run loop searching for new cars.
+     * <p>
+     * Loop is finished when there are no more new cars
+     * (objects from API are repeating with objects inside database)
+     *
+     * @param filter currently processed filter
+     * @param query final GraphQL indicating String query
+     */
+    private void loadDataForSingleFilter(Filter filter, String query) {
+        OtomotoDto otomotoDto;
+        try {
+            otomotoDto = OBJECT_MAPPER.readValue(filter.getData(), OtomotoDto.class);
+            otomotoDto.setQuery(query);
+        } catch (JsonProcessingException e) {
+            LOGGER.error("[{}][FilterId: {}] There was an error while parsing data.", TIMER_NAME, filter.getId(), e);
+            return;
+        }
+
+        if (otomotoDto.getVariables() == null) {
+            LOGGER.error("[{}][FilterId: {}] Filter has incorrect json data.", TIMER_NAME, filter.getId());
+            return;
+        }
+
+        long page = 1;
+        boolean found = false;
+        while (!found) {
+            found = finishedProcessingFilterBatchly(otomotoDto, filter, page);
+            page++;
+        }
+    }
+
+    /**
+     * Create dto with next pages and send POST request
+     *
+     * @param otomotoDto - dto sent to REST API
+     * @param filter currently processed filter
+     * @param page page number. There are 32 elements per page
+     * @return true if there are no more new elements and timer for certain filter should be finished
+     */
+    private boolean finishedProcessingFilterBatchly(OtomotoDto otomotoDto, Filter filter, Long page) {
```

Comment:
Misleading method name. This name means that the method should do nothing other than check whether filter processing is finished.

Reply:
Changed method names